src/python/pants/backend/python/subsystems/python_native_code.py

```diff
@@ -39,6 +42,17 @@ def register_options(cls, register):
     register('--native-source-extensions', type=list, default=cls.default_native_source_extensions,
              fingerprint=True, advanced=True,
              help='The extensions recognized for native source files in `python_dist()` sources.')
+    # TODO(#7735): move this to general subprocess support!
+    # NB: CPPFLAGS and LDFLAGS are necessary on certain systems (i.e. macOS) to get cryptography
```

Comment:
This feels slightly icky to hard-code into Pants, because I'm sure there are other wheels out there that will depend on other random environment settings, and it feels like this information should live in a BUILD file or in something we can derive from pypi, rather than something we hard-code. But I don't have a concrete suggestion, so...

Reply:
TensorFlow support used a Conan package initially, before #7046. It feels like `UnpackWheels` or something of the sort might be usable for wheel introspection of this type, perhaps.

Reply:
Removed the comment because I agree that the flags are useful beyond `cryptography` specifically.